### ISSUE TITLE
StrongRoute child category

### ISSUE THREAD

**Issue by Allcharles** [QutEcoacoustics/workbench-client#159]
Child categories (categories which are not a root `StrongRoute`) create duplicate routes. Currently this does not causes issues, however if there becomes a difference between the children route and parent route versions, this could cause intermittent bugs dependent on the load order of modules. The cause of this issue is that the root of the `StrongRoute` belongs to another category. Because of this, the routes for the parent categories are recalculated and added to the applications route. The solution to this is to allow a `StrongRoute` to reference another `StrongRoute` to inherit from, but not calculate from.

**Comment by atruskie** [comment]
So _categories_ are just a name and an icon. They shouldn't have routes.

The category interface overlays with a page interface. Thus a page can _act as_ a category for another page, and it will incidentally happen to have a route (because it is a page).

Why do categories need to be involved in the routing system at all?

**Comment by Allcharles** [comment]
The category is not whats involved with this process, its more so to accurately identify the StrongRoute that wraps all of a modules routes. In the case of Sites, the `sitesRoute` contains all of the routes related to the Sites Module. This "child category" normally would redefine all of the routes for the `projectsRoute` `StrongRoute` also (as that is the root `StrongRoute`), however by setting `isRoot` as shown below, it states that this is the root `StrongRoute` whilst also retaining the projects hierarchy.
```typescript
export const sitesRoute = projectMenuItem.route.add("sites", {}, true);
```

TLDR: I use the word category to identify the wrapping StrongRoute for a module which just so happens to always be defined in the category.

**Comment by atruskie** [comment]
can you provide an example for when `isRoot` would be set to false?

**Comment by Allcharles** [comment]
By default `isRoot` is always set false. It should only ever be set true once in a module, **if** the module routes are children of another module.
You can see the intended behaviour here:
```typescript
// Inherits projectMenuItem route hierarchy
export const sitesRoute = projectMenuItem.route.add("sites", {}, true);

// Normal added route to sitesRoute
export const newSiteMenuItem = MenuRoute({
  icon: defaultNewIcon,
  label: "New site",
  parent: projectMenuItem,
  predicate: isProjectOwnerPredicate,
  route: sitesRoute.add("new"),
  tooltip: () => "Create a new site"
});

// Will compile routes, beginning from the sitesRoute, not the true root which is the projectsRoute
const routes = sitesRoute.compileRoutes(GetRouteConfigForPage);
```

Calling `projectMenuItem.route.add("sites, {}, true)` is similar to writing `StrongRoute.Base` however instead of the route starting from `/`, it starts from `/projects/:projectId`.

**Comment by atruskie** [comment]
Okay, we talked about this offline.

The key feature here is supporting lazy-loaded feature modules.
When those modules are loaded, their routing system is effectively isolated from the parents routing system. That means routes inside the feature-module start from the empty base route `""`  again and this is why the `isRoot` parameter was being exposed in the relevant PR.

We shouldn't confuse our concept of a root URL (isRoot) with a new feature-module.

Our strongRoute structure should be a fully realised URL tree that can be decomposed into the angular route system - be it for all modules, for a main module and several feature modules.
